File: 11.go
package main

import "fmt"

func maxArea(height []int) int {
	l := 0
	r := len(height) - 1

	maximum := 0

	for l < r {
		h := min(height[l], height[r])
		maximum = max(maximum, h*(r-l))

		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}

	return maximum
}

func main() {
	var height = []int{1, 8, 6, 2, 5, 4, 8, 3, 7}
	res := maxArea(height)

	fmt.Println(res)
}

File: 11.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

int main()
{
}

int maxArea(int *height, int heightSize)
{
	int l = 0;
	int r = heightSize - 1;
	int max = 0;
	int hl, hr, mh;
	for (; l < r; ) {
		hl = height[l];
		hr = height[r];
		if (hl > hr) {
			mh = hr;
			r--;
		} else {
			mh = hl;
			l++;
		}
		if (mh * (r-l+1) > max) {
			max = mh * (r-l+1);
		}

	}

	return max;
}
